starter_code/src/movies.js

```diff
@@ -1,19 +1,83 @@
 /* eslint no-restricted-globals: 'off' */
 
-// Iteration 1: All rates average - Get the average of all rates with 2 decimals 
+// Iteration 1: All rates average - Get the average of all rates with 2 decimals
 
- 
-// Iteration 2: Drama movies - Get the average of Drama Movies
+function ratesAverage(movieArray) {
+  let averageRate = movieArray.reduce(function(accumulator, currentMovie) {
+    if (currentMovie.rate === "") {
+      return accumulator;
+    }
+    return (accumulator += parseFloat(currentMovie.rate));
+  }, 0);
 
+  return parseFloat((averageRate / movieArray.length).toFixed(2));
+}
 
-// Iteration 3: Ordering by duration - Order by time duration, ascending (in growing order)
+// Iteration 2: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate(movieArray) {
+  let dramaArray = movieArray.filter(function(movie) {
+    return movie.genre.indexOf("Drama") >= 0;
+  });
+  if (dramaArray.length === 0) {
+    return 0;
+  } else return ratesAverage(dramaArray);
+}
 
+// Iteration 3: Ordering by duration - Order by time duration, ascending (in growing order)
+function orderByYear(moviesArray) {
+  moviesArray.sort(function(movie1, movie2) {
+    if (parseFloat(movie1.year) < parseFloat(movie2.year)) {
+      return -1;
+    } else if (parseFloat(movie1.year) > parseFloat(movie2.year)) {
+      return 1;
+    } else {
+      return movie1.title.localeCompare(movie2.title);
+    }
+  });
+  return moviesArray;
+}
 // Iteration 4: Steven Spielberg. The best? - How many movies did STEVEN SPIELBERG direct
-
-
+function howManyMovies(movieArray) {
+  let spielbergArray = movieArray.filter(function(movie) {
+    return (
+      movie.director === "Steven Spielberg" && movie.genre.indexOf("Drama") >= 0
```

Comment:
you could use the includes method instead of indexOf then you do not need to write >= 0